sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_common.hpp

```diff
@@ -23,5 +28,158 @@
 #include "ur_level_zero_queue.hpp"
 #include "ur_level_zero_sampler.hpp"
 
+template <class To, class From> To ur_cast(From Value) {
+  // TODO: see if more sanity checks are possible.
+  assert(sizeof(From) == sizeof(To));
+  return (To)(Value);
+}
+
+// Trace an internal PI call; returns in case of an error.
+#define UR_CALL(Call)                                                          \
+  {                                                                            \
+    if (PrintTrace)                                                            \
+      fprintf(stderr, "UR ---> %s\n", #Call);                                  \
+    ur_result_t Result = (Call);                                               \
+    if (Result != UR_RESULT_SUCCESS)                                           \
+      return Result;                                                           \
+  }
+
+// Controls UR L0 calls tracing.
+enum UrDebugLevel {
+  UR_L0_DEBUG_NONE = 0x0,
+  UR_L0_DEBUG_BASIC = 0x1,
+  UR_L0_DEBUG_VALIDATION = 0x2,
+  UR_L0_DEBUG_CALL_COUNT = 0x4,
+  UR_L0_DEBUG_ALL = -1
+};
+
+// Controls Level Zero calls tracing.
+enum DebugLevel {
+  ZE_DEBUG_NONE = 0x0,
+  ZE_DEBUG_BASIC = 0x1,
+  ZE_DEBUG_VALIDATION = 0x2,
+  ZE_DEBUG_CALL_COUNT = 0x4,
+  ZE_DEBUG_ALL = -1
+};
+
+const int UrL0Debug = [] {
+  const char *ZeDebugMode = std::getenv("ZE_DEBUG");
+  const char *UrL0DebugMode = std::getenv("UR_L0_DEBUG");
+  uint32_t DebugMode = 0;
+  if (ZeDebugMode) {
+    DebugMode = std::atoi(ZeDebugMode);
+  } else if (UrL0DebugMode) {
```

Comment:
I think it would be better to follow UR_L0_DEBUG setting if both are given

Reply:
@smaslov-intel : ah ok, so UR_L0_DEBUG has higher priority. Good idea. Changing.